maplibre/src/render/stages/upload_stage.rs

```diff
@@ -223,6 +238,66 @@ impl UploadStage {
                                         &feature_metadata,
                                     );
                                 }
+                                StoredLayer::RasterLayer {
+                                    coords,
+                                    layer_name,
+                                    layer_data,
+                                } => {
+                                    #[cfg(feature = "raster")]
+                                    {
+                                        if let Initialized(raster_resources) = raster_resources {
+                                            buffer_pool.allocate_layer_raster(
+                                                queue,
+                                                *coords,
+                                                style_layer.clone(),
+                                            );
+                                            let img = image::load_from_memory(&layer_data).unwrap();
+                                            let rgba = img.to_rgba8();
```

Comment:
We are here in the render loop. Decoding of images should happen in the pipeline and not in the loop.